File: backend/Models/Job.cs
namespace CareerPilot.Api.Models;

public class Job
{
    public Guid Id { get; set; }

    public Guid UserId { get; set; }

    public required string CompanyName { get; set; }

    public required string PositionTitle { get; set; }

    public required string Description { get; set; }

    public string? Location { get; set; }

    public string? JobUrl { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedAt { get; set; }

    public User? User { get; set; }
}

File: frontend/src/types/job.ts
export interface Job {
  id: string
  companyName: string
  positionTitle: string
  description: string
  location: string | null
  jobUrl: string | null
  createdAt: string
  updatedAt: string | null
}

export interface CreateJobRequest {
  companyName: string
  positionTitle: string
  description: string
  location?: string | null
  jobUrl?: string | null
}

export interface UpdateJobRequest {
  companyName: string
  positionTitle: string
  description: string
  location?: string | null
  jobUrl?: string | null
}
